src/compiler/factory.ts

```diff
@@ -256,10 +250,14 @@ namespace ts {
     }
 
     export function createParameter(name: string | Identifier | BindingPattern, initializer?: Expression, location?: TextRange) {
+        return createParameterWithDotDotDotToken(/*dotDotDotToken*/ undefined, name, initializer, location);
+    }
+
+    export function createParameterWithDotDotDotToken(dotDotDotToken: Node, name: string | Identifier | BindingPattern, initializer?: Expression, location?: TextRange) {

```

Comment:
Just wonder should `createParameterWithDotDotDotToken` call to `createParameter` then append `node.dotDotDotTOken`? Then `createParameterWithDotDotDotToken` wouldn't have to take parameter named `dotDotDotToken`